troposphere/static/js/components/projects/resources/instance/details/actions/InstanceActionsAndLinks.jsx

```diff
@@ -93,6 +93,68 @@ export default React.createClass({
         form[0].submit();
     },
 
+    onGuacDesktop: function(ipAddr, instance) {
+        // TODO:
+        //      move this into a utilties file
+        var CSRFToken = findCookie("tropo_csrftoken");
+
+        // build a form to POST to web_desktop
+        var form = $("<form>")
+            .attr("method", "POST")
+            .attr("action", "/guacamole")
+            .attr("target", "_blank");
+
+        form.append($("<input>")
+            .attr("type", "hidden")
+            .attr("name", "ipAddress")
+            .attr("value", ipAddr));
+
+        form.append($("<input>")
+            .attr("type", "hidden")
+            .attr("name", "protocol")
+            .attr("value", "vnc"));
+
+        form.append($("<input>")
+            .attr("type", "hidden")
+            .attr("name", "csrfmiddlewaretoken")
+            .attr("style", "display: none;")
+            .attr("value", CSRFToken));
+
+        $("body").append(form);
+        form[0].submit();
+    },
+
+    onGuacShell: function(ipAddr, instance) {
+        // TODO:
+        //      move this into a utilties file
+        var CSRFToken = findCookie("tropo_csrftoken");
+
+        // build a form to POST to web_desktop
+        var form = $("<form>")
+            .attr("method", "POST")
+            .attr("action", "/guacamole")
+            .attr("target", "_blank");
+
+        form.append($("<input>")
+            .attr("type", "hidden")
+            .attr("name", "ipAddress")
+            .attr("value", ipAddr));
+
+        form.append($("<input>")
+            .attr("type", "hidden")
+            .attr("name", "protocol")
+            .attr("value", "ssh"));
+
+        form.append($("<input>")
+            .attr("type", "hidden")
+            .attr("name", "csrfmiddlewaretoken")
+            .attr("style", "display: none;")
+            .attr("value", CSRFToken));
+
+        $("body").append(form);
+        form[0].submit();
```

Comment:
This should be done via an XHR. No need to create a form, to make a network request.

See http://api.jquery.com/jQuery.post/